NewsFeed/Common/Views/ScrollableSegmentedControl.swift

```diff
@@ -0,0 +1,228 @@
+//
+//  ScrollableSegmentedControl.swift
+//  ScrollableSegmentedControl
+//
+//  Created by MArko Satlan on 22/07/2018.
+//  Copyright © 2018 MArko Satlan. All rights reserved.
+//
+
+import UIKit
+
+class ScrollableSegmentedControl: UIControl {
+
+// MARK: - Properties
+    // Open
+    open var font = UIFont(name: "Georgia", size: 15) {
+        didSet {
+            setNeedsLayout()
+        }
+    }
+    open var buttonSpacing: CGFloat = 15 {
+        didSet {
+            setNeedsLayout()
+        }
+    }
+    open var buttonMargin: CGFloat = 15 {
+        didSet {
+            setNeedsLayout()
+        }
+    }
+    open var selectedSegmentIndex: Int = -1 {
+        didSet {
+            highlightSelectedSegment()
+            calculateAndSetScrollOffset()
+      
+            // deselect previous button if different button is tapped
+            if oldValue != selectedSegmentIndex {
+                let button = buttonsArray.filter{$0.tag == oldValue}.first
+                if let previousButton = button {
```

Comment:
You can combine lines `37` and `38` into one line, like:
`if let previousButton = buttonsArray.filter{$0.tag == oldValue}.first {`

Absolutely no issue there with the code as it is, I think it's just more readable to have it as one line. Also, you can have more than one `let` in an `if` clause, which is why I prefer this way.

Reply:
![image](https://user-images.githubusercontent.com/41108005/43740700-4fdb0f5c-99cc-11e8-8dd4-6b2e48dd7d31.png)

First thing I tried, got few errors